src/NServiceBus.Core/UnitOfWork/TransactionScopes/TransactionScopeUnitOfWorkBehavior.cs

```diff
@@ -16,8 +16,7 @@ public HandlerTransactionScopeWrapperBehavior(TransactionOptions transactionOpti
         {
             if (Transaction.Current != null)
             {
-                await next().ConfigureAwait(false);
-                return;
+                throw new Exception("Ambient transaction already present, please remove the transaction scope unit of work by removing the call to .UnitOfWork().WrapHandlersInATransactionScope()");
```

Comment:
I think this is misleading. It would be better to tell the user what happened, not what to do. What happened is he configured the transport in TS mode. Tell him that these two features (transport TS and unit of work TS) can't be used together